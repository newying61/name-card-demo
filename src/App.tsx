import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
import GlobalStyle from './styled.globalstyle';
import AppHeader from './components/AppHeader';
import AppNav from './components/AppNav';
import NameCardsViewerContainer from './containers/NameCardsViewerContainer';
import { NameCardCreateForm, NameCardUpdateForm } from './containers/NameCardEditFormContainer';
import configureStore from './store/configureStore';

const store = configureStore();

const App: React.FC = () => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <>
          <GlobalStyle />
          <AppHeader />
          <AppNav />
          <Switch>
            <Route path="/cards" exact component={NameCardsViewerContainer} />
            <Route path="/cards/:id" component={NameCardUpdateForm} />
            <Route path="/add" component={NameCardCreateForm} />
            <Redirect to="/cards" />
          </Switch>
        </>
      </BrowserRouter>
    </Provider>
  );
}

export default App;
