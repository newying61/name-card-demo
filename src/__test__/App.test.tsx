import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from '../App';

afterEach(cleanup);

describe('AppHeader', () => {
    it('should render without crash', () => {
      const { container } = render(<App />);

      expect(container.firstChild).toMatchSnapshot();
  });
});