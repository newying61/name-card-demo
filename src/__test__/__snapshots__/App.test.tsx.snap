// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader should render without crash 1`] = `
<h1
  class="sc-htpNat gWfLTp"
>
  <span>
    Name cards viewer
  </span>
  <img
    alt="logo"
    aria-hidden="true"
    height="80"
    role="presentation"
    src="logo.svg"
    width="80"
  />
</h1>
`;
